core/ClassInfo.php

```diff
@@ -163,21 +153,14 @@ public static function subclassesFor($class) {
 	 * eg: self::class_name('dataobJEct'); //returns 'DataObject'
 	 *
 	 * @param string|object $nameOrObject The classname or object you want to normalise
-	 *
 	 * @return string The normalised class name
 	 */
 	public static function class_name($nameOrObject) {
 		if (is_object($nameOrObject)) {
 			return get_class($nameOrObject);
 		} elseif (!self::exists($nameOrObject)) {
-			Deprecation::notice(
-				'4.0',
-				"ClassInfo::class_name() passed a class that doesn't exist. Support for this will be removed in 4.0",
-				Deprecation::SCOPE_GLOBAL
-			);
-			return $nameOrObject;
+			throw new InvalidArgumentException("Class \"{$nameOrObject}\" doesn't exist");
```

Comment:
If you’re tidying this up, can we just remove this entire `elseif` block? `new ReflectionClass` will throw an exception below this anyway :)